src/inmanta/db/util.py

```diff
@@ -94,8 +100,46 @@ async def execute_buffer(self) -> None:
         await self.client.execute(self.commandbuffer)
         self.commandbuffer = ""
 
+    async def _parse_fq_table_name(self, fq_table_name: str) -> tuple[Optional[str], str]:
+        """
+        Parse a fully qualified PostgreSQL table name into its schema and table components.
+
+        :return: A tuple where the first element is the schema name and the second the table name.
+                 If the provided fq_table_name doesn't contain a schema, the first element in the tuple
+                 will be None.
+        """
+        if "." in fq_table_name:
+            schema, table_name = fq_table_name.split(".", maxsplit=1)
+        else:
+            schema = None
+            table_name = fq_table_name
+
+        # The schema or table name might be surrounded in quotes when the name conflicts with a keyword.
+        if schema:
+            schema = schema.strip(' "')
```

Comment:
* Normally it would not be required, but it doesn't hurt to remove leading and trailing white spaces.
* No (see: https://www.postgresql.org/docs/13/sql-syntax-lexical.html). A single quote is user for string literals.